core/payment/src/processor.rs

```diff
@@ -344,14 +344,17 @@ impl PaymentProcessor {
         let payer_addr = msg.sender;
         let payee_addr = msg.recipient;
 
-        if msg.order_ids.len() == 0 {
+        if msg.order_ids.is_empty() {
             return Err(OrderValidationError::new("order_ids is empty").into());
         }
         let orders = self
             .db_executor
             .as_dao::<OrderDao>()
             .get_many(msg.order_ids, driver.clone())
             .await?;
+        if orders.is_empty() {
+            return Err(OrderValidationError::new("orders not found in the database").into());
+        }
```

Comment:
I'd suggest moving this check into `validate_orders`. That's exactly what the function is for.

Reply:
Well, so far it validated only existing orders, so I kept the current flow. Moving it as you wish.